### PR TITLE
fix(profiling): allow to build on Alpine Linux

### LINKED ISSUE #1388: 0.37.0 fails to build on Alpine Linux 3.9
Thanks for taking the time for reporting an issue!

Before reporting an issue on dd-trace-py, please be sure to provide all
necessary information.

If you're hitting a bug, make sure that you're using the latest version of this
library.

### Which version of dd-trace-py are you using?
0.37.0
### Which version of the libraries are you using?

This was building a wheel of ddtrace only
You can copy/paste the output of `pip freeze` here.

### How can we reproduce your problem?
pip wheel ddtrace==0.37.0
### What is the result that you get?
```
 gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Os -fomit-frame-pointer -g -Os -fomit-frame-pointer -g -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100
000 -fPIC -I/usr/include/python3.6m -c ddtrace/profiling/collector/stack.c -o build/temp.linux-x86_64-3.6/ddtrace/profiling/collector/stack.o -DPy_BUILD_CORE
  ddtrace/profiling/collector/stack.c: In function '__Pyx_PyInt_As_pthread_t':
  ddtrace/profiling/collector/stack.c:8505:58: error: arithmetic on pointer to an incomplete type
       const pthread_t neg_one = (pthread_t) ((pthread_t) 0 - (pthread_t) 1), const_zero = (pthread_t) 0;
                                                            ^
  ddtrace/profiling/collector/stack.c:8428:47: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
               if (unlikely(value != (func_type) (target_type) value)) {\
                                                 ^
  ddtrace/profiling/collector/stack.c:808:43: note: in definition of macro 'unlikely'
     #define unlikely(x) __builtin_expect(!!(x), 0)
                                             ^
  ddtrace/profiling/collector/stack.c:8421:5: note: in expansion of macro '__PYX__VERIFY_RETURN_INT'
       __PYX__VERIFY_RETURN_INT(target_type, func_type, func_value, 0)
       ^~~~~~~~~~~~~~~~~~~~~~~~
  ddtrace/profiling/collector/stack.c:8526:26: note: in expansion of macro '__PYX_VERIFY_RETURN_INT'
                   case  1: __PYX_VERIFY_RETURN_INT(pthread_t, digit, digits[0])
```
### What is the result that you expected?
That it builds.

### PR DESCRIPTION
**Pull request by jd**
This hides the pthread_t type usage and force the cast to unsigned long thread
no matter what.
This is fine with glibc and musl libc at least.

Fixes #1388

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -938,6 +949,7 @@ workflows:
             - requestsgevent
             - sqlalchemy
             - sqlite3
+            - test_build_alpine
```
Comment (brettlangdon) [inline .circleci/config.yml:952]: 👍 thank you

Comment (brettlangdon) [reply .circleci/config.yml:952]: I'd consider adding all versions we support on alpine, but this is fine.

Comment (jd) [reply .circleci/config.yml:952]: Indeed, the compatibility issue is with their libc, not with any Python version specifically, so I don't think it's worth it.

**Comment by jd** [comment]
> should we add a note to our documentation "How to build from source" that at least lists the required system packages?

Do you mean for Alpine, or for every Linux distro?

**Comment by brettlangdon** [comment]
> Do you mean for Alpine, or for every Linux distro?

Yes? It should mostly just be "you need python headers and a c++ compiler", example for `apt-get install` and example of `apt add`

probably don't need to be exhaustive... ?

or maybe not worth it?

**Comment by jd** [comment]
I don't think it has been a problem so far. But if we ever get support/issues opened on this, it might be worth it.

(don't misread me, I don't want to sound like the guy who doesn't want to write documentation. 😆)

File: ddtrace/profiling/collector/stack.pyx [hunk 2]
```diff
@@ -32,8 +32,12 @@ IF UNAME_SYSNAME == "Linux":
 
     cdef extern from "<pthread.h>":
         # POSIX says this might be a struct, but CPython relies on it being an unsigned long.
-        ctypedef unsigned long pthread_t
-        int pthread_getcpuclockid(pthread_t thread, clockid_t *clock_id)
+        # We should be defining pthread_t here like this:
+        # ctypedef unsigned long pthread_t
+        # but e.g. musl libc defines pthread_t as a struct __pthread * which breaks the arithmetic Cython
+        # wants to do.
+        # We pay this with a warning at compilation time, but it works anyhow.
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/collector/stack.pyx:39]: :+1:

**Pull request merged**